src/parquet/file_writer.cc

```diff
@@ -125,21 +191,43 @@ class RowGroupSerializer : public RowGroupWriter::Contents {
   bool closed_;
   int current_column_index_;
   mutable int64_t num_rows_;
+  bool row_group_by_size_;
 
   void CheckRowsWritten() const {
-    int64_t current_rows = current_column_writer_->rows_written();
-    if (num_rows_ < 0) {
-      num_rows_ = current_rows;
-      metadata_->set_num_rows(current_rows);
-    } else if (num_rows_ != current_rows) {
-      std::stringstream ss;
-      ss << "Column " << current_column_index_ << " had " << current_rows
-         << " while previous column had " << num_rows_;
-      throw ParquetException(ss.str());
+    // verify when only one column is written at a time
+    if (!row_group_by_size_ && column_writers_.size() > 0 && column_writers_[0]) {
```

Comment:
This if-else is one of the reasons to use a new `RowGroupWriter` class in my previous version of this.